cmd/searcher/search/search_structural.go

```diff
@@ -78,9 +79,35 @@ func ToFileMatch(combyMatches []comby.FileMatch) (matches []protocol.FileMatch)
 	return matches
 }
 
+var languageExtensions = lazyregexp.New(`\.\w+`)
+
+// inferExtension converts includePatterns like "\.c$|\.cats$|\.h$|\.idc$" to
+// simply []string{".c",".cats",".h",".idc"} so that comby can infer the correct language
+// matcher. The "\.c$" or "\.go$" format arises when we specify, for example,
+// "lang:c" or "lang:go".
+func inferExtensions(includePatterns []string) (new []string) {
```

Comment:
nit: 
```suggestion
func inferExtensions(includePatterns []string) (exts []string) {
```

Because `new` is a keyword, worth avoid using it.